code/modules/clothing/neck/bodycamera.dm

```diff
@@ -0,0 +1,166 @@
+/*
+ * Bodycamera stuff
+ * NOTE: Unlike regular cameras, bodycams can only stream to one camera network at a time. Unless you want to give it more networks,
+ * the code will only ever worry about bodcam.network[1]
+ */
+
+/obj/item/clothing/neck/bodycam
+	name = "body camera"
+	desc = "A wearable camera, capable of streaming a live feed."
+	icon_state = "bodycam_off"
+	item_state = "bodycam_off"
+	var/prefix = "sec"//used for sprites, miner etc
+	var/obj/machinery/camera/bodcam = null
+	var/setup = FALSE
+	var/preset = FALSE //if true, the camera is already configured and cannot be reset
+	actions_types = list(/datum/action/item_action/toggle_bodycam)
+	strip_delay = 1 SECONDS //takes one second to strip, so a downed officer can be un-cammed quickly
+	w_class = WEIGHT_CLASS_NORMAL
+	var/mob/listeningTo //This code is simular to the code for the RCL.
+
+/obj/item/clothing/neck/bodycam/Initialize()
+	..()
+	ADD_TRAIT(src, TRAIT_NO_STORAGE, TRAIT_GENERIC)
+	bodcam = new(src)
+	bodcam.c_tag = "NT_BodyCam"
+	bodcam.network = list("ss13")
+	bodcam.internal_light = FALSE
+	bodcam.status = FALSE
+	update_icon()
+
+/obj/item/clothing/neck/bodycam/attack_self(mob/user)
+	if(!setup)
+		AltClick(user)
+		return
+	if(bodcam.status)
+		bodcam.status = FALSE
+		to_chat(user, "<span class='notice'>You shut off the body camera.</span>")
+		Screenfuzz("Error: Feed disconnected")
+		UnregisterSignal(user, COMSIG_MOVABLE_MOVED)
+		listeningTo = null
+	else
+		bodcam.status = TRUE
+		to_chat(user, "<span class='notice'>You turn on the body camera.</span>")
+		getMobhook(user)
+	update_icon()
+
+/obj/item/clothing/neck/bodycam/AltClick(mob/user)
+	if(preset)
+		return //can't change the settings on it if it's preset
+	var/name = stripped_input(user, "What would you like your camera's display name to be?", "Camera id", "[user.name]")
+	if(name)
+		bodcam.c_tag = "(Bodycam) " + name
```

Comment:
```suggestion
		bodcam.c_tag = "(Bodycam) [name]"
```
Would work too, not sure if this is required, will need alexkar to check.

Reply:
This would work fine, yeah.